### PR TITLE
Refactor gha

### PR DESCRIPTION
**Pull request by ccwienk**
First (small) refactoring - more to come.

### CODE REVIEW

File: .github/actions/gcp-auth/action.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
+name: Authenticate against GAR
+description: |
+  Authenticates against GCP using OIDC. This action defines a default Service-Account as
+  identity.
+inputs:
+  gcp_service_account:
+    description: The Service-Account to impersonate
+    required: false
+    default: ocm-ci-opensource@gardener-project.iam.gserviceaccount.com
```
Comment (zkdev) [inline .github/actions/gcp-auth/action.yaml:9]: this one is only used within ocm org, rather use `ci-gcr-operator@gardener-project.iam.gserviceaccount.com` as default value

File: .github/actions/gcp-auth/action.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
+name: Authenticate against GAR
```
Comment (zkdev) [inline .github/actions/gcp-auth/action.yaml:1]: there is an inconsistency, `gcp-auth` vs `Authenticate against GAR`
is this action only working for GAR, or for GCP in general?

**Review by zkdev — CHANGES_REQUESTED** [review]

File: .github/actions/gcp-auth/action.yaml (hunk 1 above)
Comment (ccwienk) [reply .github/actions/gcp-auth/action.yaml:9]: hmm, valid point. Maybe even better yet, we might create a dedicated service-account just for OIDC / GHAs

File: .github/actions/gcp-auth/action.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      The hostname of the GAR (for generating docker-cfg). For convenience, it is also
+      okay to pass a full image-reference (in which case hostname is parsed from parsed value).
+      Note that no special-handling for docker-hub (e.g. `alpine` ->
+      registry-1.docker.io/library/alpine:latest) is done.
+    required: false
+    default: europe-docker.pkg.dev
+outputs:
+  access_token:
+    description: The obtained access-token
+    value: ${{ steps.authenticate.outputs.access_token }}
```
Comment (8R0WNI3) [inline .github/actions/gcp-auth/action.yaml:31]: I'm wondering, how is this working? I'm not seeing any step with id `authenticate`, and the "Export Access-Token" step does not define any id 🤔

Comment (ccwienk) [reply .github/actions/gcp-auth/action.yaml]: @8R0WNI3 : good catch - fixed

**Review by zkdev — APPROVED** [review]
/lgtm

**Pull request merged**